src/controller/python/chip-device-ctrl.py

```diff
@@ -816,6 +816,51 @@ def zcl_command(
     except Exception as e:
         return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e))
 
+def zcl_read_attribute(
+        cluster: str,
+        attribute: str,
+        node_id: int,
+        endpoint_id: Optional[int] = 0,
+        group_id: Optional[int] = 0):
+    """Generic RPC call to read ZCL cluster attributes
+    zclread command format: zclread <Cluster> <Attribute> <NodeId> <EndpointId> <GroupId> [arguments]
+    
+    Args:
+        cluster (str): Name of cluster
+        command (str): Command to be run for the mentioned cluster
+        node_id (int): node_id assigned to the DUT
+        endpoint_id (int): Optional, endpoint id
+        group_ip (int): Optional, group id
+
+    Raises:
+        exceptions.UnknownCommand: when incorrect cluster and/or command passed
+    
+    Returns:
+        Dict[str, Any]: Dictionary of RPC response for ZCL read
+    """
+    try:
+        all_commands = device_manager.devCtrl.ZCLCommandList()
+        all_attrs = device_manager.devCtrl.ZCLAttributeList()
+        if cluster not in all_commands or attribute not in all_attrs:
+            raise exceptions.UnknownCluster(cluster)
+        else:
+            response = device_manager.devCtrl.ZCLReadAttribute(
+                    cluster=cluster,
+                    attribute=attribute,
+                    nodeid=node_id,
+                    endpoint=endpoint_id,
+                    groupid=group_id,
+                    blocking=True)
+            if response.status:
+                return __get_response_dict(status = StatusCodeEnum.FAILED)
+            if response:
+                return __get_response_dict(status = StatusCodeEnum.SUCCESS, result = str(response.value))
+            else:
+                return __get_response_dict(status = StatusCodeEnum.SUCCESS)
+    except exceptions.ChipStackException as ex:
+        print("An exception occurred during reading ZCL attribute:")
+        print(str(ex))
```

Comment:
why are you not raising this exception? I see other parts of this file do similar thing but why? Isn't better to let someone above handle it or suppress it if they want to. This is a dangerous behavior is all, it will lead to silent failures. 

Also just in general, if you want to be logging things I would suggest using `logging` or something similar as opposed to print https://docs.python.org/3/howto/logging.html#when-to-use-logging

Reply:
Updated

Reply:
sgtm